src/controllers/movies.js

```diff
@@ -0,0 +1,164 @@
+const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+
+const getAllMovies = async (req, res) => {
+  console.log("get movies");
+
+  if (!req.query.runtimeLt && !req.query.runtimeGt) {
+    const movies = await prisma.movie.findMany({
+      include: {
+        screenings: true,
+      },
+    });
+
+    res.status(200).json({
+      movies,
+    });
+  }
+
+  if (req.query.runtimeLt) {
+    const movies = await prisma.movie.findMany({
+      where: {
+        runtimeMins: {
+          lt: Number(req.query.runtimeLt),
+        },
+      },
+    });
+
+    res.status(200).json({
+      movies,
+    });
+  }
+
+  if (req.query.runtimeGt) {
+    const movies = await prisma.movie.findMany({
+      where: {
+        runtimeMins: {
+          gt: Number(req.query.runtimeGt),
+        },
+      },
+    });
+
+    res.status(200).json({
+      movies,
+    });
+  }
+
+  if (req.query.runtimeLt && req.query.runtimeGt) {
+    // + try
+    const movies = await prisma.movie.findMany({
+      where: {
+        AND: [
+          {
+            runtimeMins: {
+              lt: req.query.runtimeLt,
+            },
+          },
+          {
+            runtimeMins: {
+              gt: req.query.runtimeGt,
+            },
+          },
+        ],
+      },
+    });
+
+    res.status(200).json({
+      movies,
+    });
+  }
+};
+
+const createNewMovie = async (req, res) => {
+  console.log("CREATING");
+  const { title, runtimeMins } = req.body;
+
+  if (!title || !runtimeMins) {
+    return res.status(400).json({
+      error: "Missing fields in request body",
+    });
+  }
+
+  try {
+    console.log('req', req.body.title);
+    const foundMovie = await prisma.movie.findFirst({
+      where: {
+        title: req.body.title
+      }
+    }) 
+    console.log('found', foundMovie);
+    if (foundMovie === null) {
+      const createdMovie = await prisma.movie.create({
+        data: {
+          title,
+          runtimeMins,
+        },
+      });
+      res.status(201).json({ movie: createdMovie });
+    } else if (foundMovie !== null) {
+        return res.status(409).json({ error: "This Movie Already Exists" });
+    }
```

Comment:
You can eliminate the `else if` here by adding a return at the end of your first if statement:

```js
if (foundMovie === null) {
      const createdMovie = await prisma.movie.create({
        data: {
          title,
          runtimeMins,
        },
      });
      
      return res.status(201).json({ movie: createdMovie });
}

return res.status(409).json({ error: "This Movie Already Exists" });
```